src/OpenConext/ProfileBundle/Storage/SingleCookieStorage.php

```diff
@@ -42,13 +43,43 @@ class SingleCookieStorage implements EventSubscriberInterface
      */
     private $cookieValue;
 
-    public function __construct($cookieDomain, $cookieKey)
-    {
+    /**
+     * @var DateTime|0
+     */
+    private $cookieExpirationDate;
+
+    /**
+     * @var boolean
+     */
+    private $cookieSecure;
+
+    /**
+     * @var boolean
+     */
+    private $cookieHttpOnly;
+
+    public function __construct(
+        $cookieDomain,
+        $cookieKey,
+        DateTime $cookieExpirationDate = null,
+        $cookieSecure = true,
+        $cookieHttpOnly = true
+    ) {
         Assert::string($cookieDomain);
         Assert::string($cookieKey);
+        Assert::boolean($cookieSecure);
+        Assert::boolean($cookieHttpOnly);
```

Comment:
would add some specific messages "Expected value to be a boolean, NULL given" for instance is not really helpful when there are multiple boolean checks.